configure.py

```diff
@@ -2175,4 +2176,7 @@ def make_bin_override():
 
 print_verbose("running: \n    " + " ".join(['python', 'tools/gyp_node.py'] + gyp_args))
 run_gyp(gyp_args)
+if sys.platform == 'win32':
+  os.path.isfile('./compile_commands.json') and os.unlink('./compile_commands.json')
+  shutil.copy2('./out/' + config['BUILDTYPE'] + '/compile_commands.json', './compile_commands.json')
```

Comment:
Why do we copy in here, but not in line 2168?

Reply:
Because symlinks in Windows are troublesome and require special configuration (e.g. Developer Mode), so I just went and copied instead of symlinked on Windows. On non-Windows, I kept the symlink as is rather than copying.

Reply:
So, every build needs to copy this. Previously having a symlink doesn't require the same operation to be done. Can we verify if incremental builds reflect the same output?

cc @nodejs/platform-windows

Reply:
The `compile_commands.json` likely only depends on the configuration flags passed to GYP, as it is generated by it. Since the flags can change, I opted to copying it each time to be safe and to be sure it is correct, I don't think the file is that big that it is noticeable in contrast to all the other things that are already done on each build.